lib/filterable_sortable.rb

```diff
@@ -10,18 +10,39 @@ module FilterableSortable
       if filter[:search]
         search(filter[:search])
       elsif filter[:custom] && filter[:custom] != 'all'
-        self.send(filter[:custom].to_sym) if self.methods.include?(filter[:custom].to_sym)
+        if self.methods.include?(filter[:custom].to_sym) &&
+          !(filter[:custom] =~ /delete_|update_|create|drop_|destroy_|\!/)
+          self.readonly.send(filter[:custom].to_sym)
+        end
       end
     }
 
     scope :search, lambda { |term|
-      conditions = self.columns_hash.collect{|k,v| k unless v.type.in?([:datetime, :time, :date])}.compact.collect{|f| "#{self.table_name}.#{f} like '%#{term}%'"}.join(' OR ')
+      conditions = self
+                     .columns_hash
+                     .collect { |k, v| k unless v.type.in?([:datetime, :time, :date]) }
+                     .compact
+                     .collect { |f| "#{self.table_name}.#{f} like #{ActiveRecord::Base::sanitize("'%#{term}%'")}" }
+                     .join(' OR ')
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align `.join` with `self` on line 21.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.